OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -59,70 +60,42 @@ public Move(Actor self, CPos destination, int nearEnough)
 
 		public Move(Actor self, CPos destination, WRange nearEnough)
 		{
-			mobile = self.Trait<Mobile>();
-			moveDisablers = self.TraitsImplementing<IDisableMove>();
+			Init(self, destination, nearEnough);
 
 			getPath = () => self.World.WorldActor.Trait<IPathFinder>()
 				.FindUnitPath(mobile.ToCell, destination, self);
-			this.destination = destination;
-			this.nearEnough = nearEnough;
-		}
-
-		public Move(Actor self, CPos destination, SubCell subCell, WRange nearEnough)
-		{
-			mobile = self.Trait<Mobile>();
-			moveDisablers = self.TraitsImplementing<IDisableMove>();
-
-			getPath = () => self.World.WorldActor.Trait<IPathFinder>()
-				.FindUnitPathToRange(mobile.FromCell, subCell, self.World.Map.CenterOfSubCell(destination, subCell), nearEnough, self);
-			this.destination = destination;
-			this.nearEnough = nearEnough;
 		}
 
 		public Move(Actor self, CPos destination, Actor ignoredActor)
 		{
-			mobile = self.Trait<Mobile>();
-			moveDisablers = self.TraitsImplementing<IDisableMove>();
+			Init(self, destination, WRange.Zero);
 
 			getPath = () =>
 				self.World.WorldActor.Trait<IPathFinder>().FindPath(
 					PathSearch.FromPoint(self.World, mobile.Info, self, mobile.ToCell, destination, false)
 					.WithIgnoredActor(ignoredActor));
 
-			this.destination = destination;
-			this.nearEnough = WRange.Zero;
 			this.ignoredActor = ignoredActor;
 		}
 
-		public Move(Actor self, Target target, WRange range)
+		public Move(Actor self, Func<List<CPos>> getPath)
 		{
-			mobile = self.Trait<Mobile>();
-			moveDisablers = self.TraitsImplementing<IDisableMove>();
-
-			getPath = () =>
-			{
-				if (!target.IsValidFor(self))
-					return NoPath;
-
-				return self.World.WorldActor.Trait<IPathFinder>().FindUnitPathToRange(
-					mobile.ToCell, mobile.ToSubCell, target.CenterPosition, range, self);
-			};
-
-			destination = null;
-			nearEnough = range;
+			Init(self, null, WRange.Zero);
+			this.getPath = getPath;
 		}
 
-		public Move(Actor self, Func<List<CPos>> getPath)
+		private void Init(Actor self, CPos? destination, WRange nearEnough)
```

Comment:
Looks like you could make this a private ctor and chain calls to it, which would allow you to keep the readonly modifiers you had to drop.